presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -858,24 +859,44 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
             }
 
             List<TypeSignatureProvider> argumentTypes = getCallArgumentTypes(node.getArguments(), context);
-            Signature function = resolveFunction(node, argumentTypes, metadata);
+
+            Signature function;
+            try {
+                function = metadata.resolveFunction(node.getName(), argumentTypes);
+            }
+            catch (PrestoException e) {
+                if (e.getLocation().isPresent()) {
+                    // If analysis of any of the argument types (which is done lazily to deal with lambda expressions) fails, we
```

Comment:
Nit: these comments would look better if the text was balanced over the two lines